### PR TITLE
init typify-cli

### PR DESCRIPTION
**Pull request by AlbanMinassian**
help anybody to view code generate quickly

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 	"example-build",
 	"example-macro",
 ]
 
-default-members = [
-	"typify",
-	"typify-macro",
-	"typify-impl",
-]
+default-members = ["typify", "typify-cli", "typify-macro", "typify-impl"]
```
Comment (ahl) [inline Cargo.toml:11]: is there a reason this is all on one line?

File: typify-cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "typify-cli"
+version = "0.1.0"
```
Comment (ahl) [inline typify-cli/Cargo.toml:3]: can you please make this match the version in all the other crates?

File: typify-cli/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+[[bin]]
+name = "typify-cli"
+path = "src/main.rs"
+
+[dependencies]
+serde = "1.0"
+schemars = "0.8"
+serde_json = "1.0"
+typify = { path = "../typify" }
```
Comment (ahl) [inline typify-cli/Cargo.toml:16]: can you add the version here as well please?

File: typify-cli/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+use schemars::schema::Schema;
+use std::env;
+use typify::TypeSpace;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    if args.len() != 2 {
+        panic!("{} need 1 argument", env!("CARGO_PKG_NAME"))
```
Comment (ahl) [inline typify-cli/src/main.rs:8]: perhaps a gentler failure mode and usage message?

**Review by ahl — COMMENTED** [review]
Thanks for doing this. I left a couple of small notes. I think it would be good to have a test of the command as well if you'd be amenable to writing that.

File: Cargo.toml (hunk 1 above)
Comment (AlbanMinassian) [reply Cargo.toml:11]: 'cargo fmt' return this

File: typify-cli/Cargo.toml (hunk 3 above)
Comment (AlbanMinassian) [reply typify-cli/Cargo.toml]: done

File: typify-cli/src/main.rs (hunk 4 above)
Comment (AlbanMinassian) [reply typify-cli/src/main.rs]: done

**Comment by AlbanMinassian** [comment]
sorry i have not time to help you more.

**Comment by ahl** [comment]
> sorry i have not time to help you more.

No problem. I'll be happy to pick this up a bit later on. I'm going to close it for now. Thanks for your contribution.

**Pull request closed without merging**